### PR TITLE
Rework back to dashboard e2e test to make it more real world and less flaky

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

During backporting we met the problem with current e2e tests, we verified correctness of the logic and reworked the test to make it less flaky and more real world.

I had to reorder tests in the second describe block as order of the tests matters and last test with slow card has a request that finishes after the test is completed, which leads to the falsy result on http request interception

seems this issue in cypress is related https://github.com/cypress-io/cypress/issues/20397

### How to verify

e2e tests should pass here and in the backport PR https://github.com/metabase/metabase/pull/32239

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js
Comment (nemanjaglumac) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:301]: How can this be 2? We've just loaded the dashboard so at this point there was only one request.

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js
Comment (nemanjaglumac) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:306]: Unless you explicitly wait for the response, you can never be sure that visiting the question will NOT interrupt and cancel this dashCardQuery request.

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js
Comment (nemanjaglumac) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:301]: The reason you're seeing this locally when you isolate the common describe block using `.only` is that the previous test leaks data.

Since the previous test never had the chance to load dashboard card before finishing, it registered the response in the next one. This might be a Cypress bug, or maybe they even intentionally set it up like that... it doesn't matter.

We should never ever adjust tests (expected Metabase behavior) based on how a certain library (or our environment) behaves.

Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:301]: > We should never ever adjust tests (expected Metabase behavior) based on how a certain library (or our environment) behaves.

I totally agree with this!

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:306]: we explicitly cancel some requests (afaik it can be done to unblock BE when it makes a heavy query), so I expected from cypress it catches all the fired http requests and it's what I wanted to test

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js]: should be good now

**Review by nemanjaglumac — APPROVED** [review]
I'm not such a huge fan of reordering these tests and relying on the test order in the future, but let's leave that for some other time.

What matters is this
image

**Review by deniskaber — APPROVED** [review]

**Pull request merged**
